kibana.yml

```diff
@@ -32,6 +32,8 @@
 # is proxied through the Kibana server.
 # elasticsearch.username: "user"
 # elasticsearch.password: "pass"
+%ELASTICSEARCH_USERNAME%
```

Comment:
But in this case you're not directly replacing the placeholder `%ELASTICSEARCH_USERNAME%` with the value of the env var `ELASTICSEARCH_USERNAME` – you're deriving a new output from the original env var which gets put here, as Vaidas pointed out above.